src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,54 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final int SUPPLY_TYPE = 0;
+    private static final int BUY_TYPE = 1;
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+
+    private int supply = 0;
+    private int buy = 0;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        supply = 0;
+        buy = 0;
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                processLine(line);
+            }
+            int result = supply - buy;
+            writeResult(toFileName, supply, buy, result);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read data from the file", e);
+        }
```

Comment:
Make getStatistic consists of 3 code lines
```suggestion
    String dataFromFile = readFile(fromFileName);
    String report = createReport(dataFromFile);
    writeToFile(report, toFileName);
```